### DISCUSSION TITLE
state: 19     USB_DETACHED_SUBSTATE_ILLEGAL

### DISCUSSION THREAD

**Discussion by fabiendostie** [marcel-licence/esp32_usb_midi#9] [Q&A] [answered]
Hello Marcel,
I've finally succeeded to install everything... i think... but i've got an error concerning usb-midi in the serial monitor.

any idea what could be causing this?

I'm on a esp32 A1S audio kit v2.2 (3378) with the 8388chip

```
esp32_basic_synth  Copyright (C) 2021  Marcel Licence
This program comes with ABSOLUTELY NO WARRANTY;
This is free software, and you are welcome to redistribute it
under certain conditions; 
Initialize Synth Module
Initialize I2S Module
Connect to ES8388 codec... 0x00: 0x05
0x01: 0x40
AdcCh1!
MixChAMPL!
ES8388 setup finished!
Reg 0x00 = 0x05
Reg 0x01 = 0x40
Reg 0x02 = 0x00
Reg 0x03 = 0x00
Reg 0x04 = 0x3c
Reg 0x05 = 0x00
Reg 0x06 = 0x00
Reg 0x07 = 0x7c
Reg 0x08 = 0x00
Reg 0x09 = 0x88
Reg 0x0a = 0x50
Reg 0x0b = 0x02
Reg 0x0c = 0x0c
Reg 0x0d = 0x02
Reg 0x0e = 0x30
Reg 0x0f = 0x20
Reg 0x10 = 0x00
Reg 0x11 = 0x00
Reg 0x12 = 0x16
Reg 0x13 = 0xb0
Reg 0x14 = 0x32
Reg 0x15 = 0x06
Reg 0x16 = 0x00
Reg 0x17 = 0x18
Reg 0x18 = 0x02
Reg 0x19 = 0x22
Reg 0x1a = 0x00
Reg 0x1b = 0x02
Reg 0x1c = 0x08
Reg 0x1d = 0x00
Reg 0x1e = 0x1f
Reg 0x1f = 0xf7
Reg 0x20 = 0xfd
Reg 0x21 = 0xff
Reg 0x22 = 0x1f
Reg 0x23 = 0xf7
Reg 0x24 = 0xfd
Reg 0x25 = 0xff
Reg 0x26 = 0x1b
Reg 0x27 = 0xb8
Reg 0x28 = 0x28
Reg 0x29 = 0x28
Reg 0x2a = 0xb8
Reg 0x2b = 0x80
Reg 0x2c = 0x00
Reg 0x2d = 0x00
Reg 0x2e = 0x1e
Reg 0x2f = 0x1e
Reg 0x30 = 0x1e
Reg 0x31 = 0x1e
Reg 0x32 = 0x00
Reg 0x33 = 0xaa
Reg 0x34 = 0xaa
rev: 15329, 15329
Reverb is ready!
Setup Serial2 with 31250 baud with rx: 21 only
Setup MidiPort2 using Serial2
ESP.getFreeHeap() 69140
ESP.getMinFreeHeap() 69120
ESP.getHeapSize() 217356
ESP.getMaxAllocHeap() 37412
Total heap: 217356
Free heap: 69140
Total PSRAM: 0
Free PSRAM: 0
Firmware started successfully
Hello now we can start

Usb init done!

state: 19
    USB_DETACHED_SUBSTATE_ILLEGAL
```

thank you for your time and code.

**Comment by marcel-licence** [comment]
I am not sure what causing this. Did you ensure that there are no PIN conflicts? In addition to that does the USB example from the library work?

**Comment by fabiendostie** [comment]
there is nothing else plugged on  it at all!
which example are you referring to, i'm using your esp32_basic_synth , everything compiles and uploads well except at the last second... i get this error... after the Firmware started successfully, Hello now we can start, Usb init done!
It's really frustrating.
the right board is selected, (#define BOARD_ESP32_AUDIO_KIT_ES8388 ) I got all the right libraries, it compiles...

can you confirm that I should be able to use usb midi straight from the esp32A1S audio kit 2.2?
without the need for anything else than the board and a usb cable pluged in a computer ?

controller > usb midi > mac   &  esp32_basic_synth on the ESP32A1S > usb midi > mac ?

Thanks

**Reply by marcel-licence** [reply]
The USB port auf the audio kit is directly connected to the USB-Serial ic. That means the audio kit does not support any other USB connections by itself. I will add a comment to the USB functionality which requires the external USB host module connected via SPI ref: https://youtu.be/Mt3rT-SVZww

**Comment by fabiendostie** [comment, marked as answer]
Thats what I thought, Thanks!
I actually made a din midi in with a 1H11L optocoupler and i see one on the led's blinking very lightly when i press keys on my midi controller. BUUUUUUT, I still hear no sound when i plug some headphones on the earphones in... Any cues as to what is happening or why I wouldn't hear any sounds.

Obviously no errors on happened while compiling, but theres no way to verify anything after the code is uploaded it's quite frustrating, I wish there was a way to see what is happening on the board once your code is uploaded!

Could you confirm that I should use IO22 as RX for midi din?

if you can point me to something that could help me finally be able to produce sounds from this I would  be SOOOOOOO grateful, It's been a hard and long process for me.

I learned a lot during the process though.

i'm just missing the final little bit. 

Thanks in advance
![IMG_20220114_070054](https://user-images.githubusercontent.com/53846121/149513149-0f12a395-5864-4f42-927c-4a7158cb2ec8.jpg)
.

**Reply by marcel-licence** [reply]
Looks good. And you also put the diode into your circuit for ESD protection. I may be IO21. Please connect a serial terminal to your serial port (or use the serial monitor of Arduino). You will get the information of the used port in Arduino. You can also ensure that the NoteOn will be called in your setup routine. In that case you should get audio indepent from your MIDI input.
